src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -691,39 +739,20 @@ def auto_transfer(self, service_id: uuid.UUID) -> inmanta_lsm.model.ServiceInsta
             transfer_type=inmanta_lsm.const.TransferTrigger.AUTO,
         )
 
-        def next_state(state: str) -> None:
-            """
-            Apply this state to our service, if it is different from the current
-            state, also increment the version
-
-            :param state: The new state to apply
-            """
-            if service.state == state:
-                return
-
-            service.last_updated = datetime.datetime.now()
-            service.version += 1
-            service.state = state
-
         try:
-            # Trigger a compile for the transition
             LOGGER.info(
                 "Triggering compile on state %s before auto transfer (%s) for service %s (%s)",
                 service.state,
                 transfer.description,
                 service.id,
                 service.service_entity,
             )
-            self.compile(
-                service_id=service.id,
-                validation=transfer.validate_,
-            )
             perform_attribute_operation(service, transfer.target_operation)
-            next_state(transfer.target)
+            self.transfer_to_next_state(service=service, transfer=transfer, is_error_transition=False)
         except Exception:
```

Comment:
This logic is broken, the exception we catch here is from the state afther the transition, nto the one of the current state